tests/test_cache.py

```diff
@@ -1,504 +1,372 @@
+def test_all_methods(field, method, mock_session):
+    """Test all relevant combinations of methods and data fields. Requests with different request
+    params, data, or json should be cached under different keys.
+    """
+    for params in [{'param_1': 1}, {'param_1': 2}, {'param_2': 2}]:
+        assert mock_session.request(method, MOCKED_URL, **{field: params}).from_cache is False
+        assert mock_session.request(method, MOCKED_URL, **{field: params}).from_cache is True
+
+
+@pytest.mark.parametrize('method', ALL_METHODS)
+@pytest.mark.parametrize('field', ['params', 'data', 'json'])
+def test_all_methods__ignore_parameters(field, method, mock_session):
+    """Test all relevant combinations of methods and data fields. Requests with different request
+    params, data, or json should not be cached under different keys based on an ignored param.
+    """
+    mock_session.cache._ignored_parameters = ['ignored']
+    params_1 = {'ignored': 1, 'not ignored': 1}
+    params_2 = {'ignored': 2, 'not ignored': 1}
+    params_3 = {'ignored': 2, 'not ignored': 2}
+
+    assert mock_session.request(method, MOCKED_URL, **{field: params_1}).from_cache is False
+    assert mock_session.request(method, MOCKED_URL, **{field: params_1}).from_cache is True
+    assert mock_session.request(method, MOCKED_URL, **{field: params_2}).from_cache is True
+    mock_session.request(method, MOCKED_URL, params={'a': 'b'})
+    assert mock_session.request(method, MOCKED_URL, **{field: params_3}).from_cache is False
+
+
+# TODO: mock response with cookies
+def test_cookies(mock_session):
+    def get_json(url):
+        return json.loads(mock_session.get(url).text)
+
+    response_1 = get_json(httpbin('cookies/set/test1/test2'))
+    with mock_session.cache_disabled():
+        assert get_json(httpbin('cookies')) == response_1
+    # From cache
+    response_2 = get_json(httpbin('cookies'))
+    assert response_2 == get_json(httpbin('cookies'))
+    # Not from cache
+    with mock_session.cache_disabled():
+        response_3 = get_json(httpbin('cookies/set/test3/test4'))
+        assert response_3 == get_json(httpbin('cookies'))
+
+
+# TODO: mock response with gzip-compressed content
+def test_gzip(mock_session):
+    assert mock_session.get(httpbin('gzip')).from_cache is False
+    assert mock_session.get(httpbin('gzip')).from_cache is True
+
+
+def test_https(mock_session):
+    assert mock_session.get(MOCKED_URL_HTTPS, verify=True).from_cache is False
+    assert mock_session.get(MOCKED_URL_HTTPS, verify=True).from_cache is True
+
+
+def test_json(mock_session):
+    assert mock_session.get(MOCKED_URL_JSON).from_cache is False
+    response = mock_session.get(MOCKED_URL_JSON)
+    assert response.from_cache is True
+    assert response.json()['message'] == 'mock json response'
+
+
+# TODO: Create mock response with redirect history
+@pytest.mark.skip(reason='httpbin.org/relative-redirect no longer returns redirects')
```

Comment:
Not important right now, but the docker image returns redirects, so it should be possible to test it against a live system:
```console
$ curl -i localhost:80/relative-redirect/2
HTTP/1.1 302 FOUND
Server: gunicorn/19.9.0
Date: Sun, 21 Mar 2021 08:50:05 GMT
Connection: keep-alive
Content-Type: text/html; charset=utf-8
Content-Length: 0
Location: /relative-redirect/1
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
```

Reply:
Nice! I made a note of that on issue #169